### PR TITLE
Interactive Conversions WIP

### PR DESCRIPTION
**Pull request by mintsoft**
Definitely a _work in progress_ !

Branched from the `mintsoft/conversions/template branch` used in #2044; this currently 'works' by bodging the current live API; in https://github.com/duckduckgo/zeroclickinfo-goodies/pull/2044 I've added some info to the API which will make this much easier to write and so forth. So ideally I think we need to get #2044 live before I can tidy up this implementation and the tests would pass etc.

I'm basically going to hard-code loads of stuff into the templates here initially and then see how I can get it datadriven nicely.

---

https://duck.co/ia/view/conversions

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by jagtalon** [comment]
:heart:

File: share/goodie/conversions/conversions.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+		force: {left: 'newton', right: 'pounds force'},
+		temperature: {left: 'celsius', right:'kelvin'},
+		digital: {left:' kibibyte', right:'byte'}
+	};
+	
+	var updateGui = function(){
+		var query = $("#zci__conversions-left-in").val() + " " + $("#zci__conversions-left-unit").val() + " in " + $("#zci__conversions-right-unit").val();
+				
+		$.getJSON("https://crossorigin.me/"+"https://api.duckduckgo.com/?format=json&q="+query, function(data){
+//		$.getJSON("https://api.duckduckgo.com/?format=json&q="+query, function(data){
```
Comment (GuiltyDolphin) [inline share/goodie/conversions/conversions.js]: @mintsoft Might want `encodeURIComponent(query)` (bit me!)

**Comment by moollaza** [comment]
@mintsoft are you interested in picking this up again?

Now that we have Math.js available we can use it instead of the API to calculate results 👍

**Comment by mintsoft** [comment]
@moollaza Yeah I'll throw the test suite at Math.js and see how much massaging it requires to make it work with our query patterns. We might as well close this for now

**Pull request closed without merging**